pootle/apps/pootle_store/unit/filters.py

```diff
@@ -6,6 +6,139 @@
+
+    def filter_checks(self):
+        if self.checks:
+            return self.qs.filter(
+                qualitycheck__false_positive=False,
+                qualitycheck__name__in=self.checks).distinct()
+        elif self.category:
+            return self.qs.filter(
+                qualitycheck__false_positive=False,
+                qualitycheck__category=self.category).distinct()
+        return self.qs.none()
+
+
+class UnitStateFilter(UnitFilter):
+    """Filter a Unit qs based on unit state"""
+
+    def filter_translated(self):
+        return self.qs.filter(state=TRANSLATED)
+
+    def filter_untranslated(self):
+        return self.qs.filter(state=UNTRANSLATED)
+
+    def filter_fuzzy(self):
+        return self.qs.filter(state=FUZZY)
+
+    def filter_incomplete(self):
+        return self.qs.filter(
+            Q(state=UNTRANSLATED) | Q(state=FUZZY))
+
+
+class UnitContributionFilter(UnitFilter):
+    """Filter a Unit qs based on user contributions"""
+
+    def __init__(self, qs, *args, **kwargs):
+        self.qs = qs
+        self.user = kwargs.get("user")
+
+    def filter_suggestions(self):
+        return self.qs.filter(
+            suggestion__state=SuggestionStates.PENDING).distinct()
+
+    def filter_my_suggestions(self):
+        if not self.user:
+            return self.qs.none()
+        return self.qs.filter(
+            suggestion__state=SuggestionStates.PENDING,
+            suggestion__user=self.user).distinct()
+
+    def filter_user_suggestions(self):
+        return self.filter_my_suggestions()
+
+    def filter_user_suggestions_accepted(self):
+        if not self.user:
+            return self.qs.none()
+        return self.qs.filter(
+            suggestion__state=SuggestionStates.ACCEPTED,
+            suggestion__user=self.user).distinct()
+
+    def filter_user_suggestions_rejected(self):
+        if not self.user:
+            return self.qs.none()
+        return self.qs.filter(
+            suggestion__state=SuggestionStates.REJECTED,
+            suggestion__user=self.user).distinct()
+
+    def filter_my_submissions(self):
+        if not self.user:
+            return self.qs.none()
+        return self.qs.filter(
+            submission__submitter=self.user,
+            submission__type__in=SubmissionTypes.EDIT_TYPES).distinct()
+
+    def filter_user_submissions(self):
+        return self.filter_my_submissions()
+
+    def filter_my_submissions_overwritten(self):
+        if not self.user:
+            return self.qs.none()
+        qs = self.qs.filter(
+            submission__submitter=self.user,
+            submission__type__in=SubmissionTypes.EDIT_TYPES)
+        return qs.exclude(submitted_by=self.user).distinct()
+
+    def filter_user_submissions_overwritten(self):
+        return self.filter_my_submissions_overwritten()
+
+
+class UnitSearchFilter(object):
+
+    filters = (
+        UnitChecksFilter, UnitStateFilter, UnitContributionFilter)
```

Comment:
This was me finding harder to compare previous code with newer code since the order was altered. Once this lands it makes no sense which is the order in where stuff is. So this is not actually about this specific line but about all the code above.